pkg/image/filefmt_test.go

```diff
@@ -46,118 +40,72 @@ func TestHeader_Match(t *testing.T) {
+		h := Header{
+			Format:      fields.Format,
+			magicNumber: fields.magicNumber,
+			mgOffset:    fields.mgOffset,
+			SizeOff:     fields.SizeOff,
+			SizeLen:     fields.SizeLen,
+		}
+		got := h.Match(args.b)
+		Expect(got).To(Equal(want))
+	},
+		table.Entry("match gz",
+			fields{"gz", []byte{0x1F, 0x8B}, 0, 0, 0},
+			args{[]byte{0x1F, 0x8B}},
+			true),
+		table.Entry("match qcow2",
+			fields{"qcow2", []byte{'Q', 'F', 'I', 0xfb}, 0, 24, 8},
+			args{[]byte{'Q', 'F', 'I', 0xfb}},
+			true),
+		table.Entry("match tar",
+			fields{"tar", []byte{0x75, 0x73, 0x74, 0x61, 0x72, 0x20}, 0x101, 124, 8},
+			args{tarbyte},
+			true),
+		table.Entry("match xz",
+			fields{"xz", []byte{0xFD, 0x37, 0x7A, 0x58, 0x5A, 0x00}, 0, 0, 0},
+			args{[]byte{0xFD, 0x37, 0x7A, 0x58, 0x5A, 0x00}},
+			true),
+		table.Entry("failed match",
+			fields{"gz", []byte{0x1F, 0x8B}, 0, 0, 0},
+			args{[]byte{'Q', 'F', 'I', 0xfb}},
+			false),
+	)
 
-	//tar bytes and offset
-	token := make([]byte, 20)
+	tokenQcow := make([]byte, 20)
 	qcowMagic := []byte{'Q', 'F', 'I', 0xfb}
 	qcowSize := []byte("10561056")
-	rand.Read(token)
-	qcowbyte := append(qcowMagic, token...)
+	rand.Read(tokenQcow)
+	qcowbyte := append(qcowMagic, tokenQcow...)
 	qcowbyte = append(qcowbyte, qcowSize...)
 
-	type args struct {
-		b []byte
-	}
-	tests := []struct {
-		name    string
-		fields  fields
-		args    args
-		want    int64
-		wantErr bool
-	}{
-		{
-			name:    "get size of qcow2",
-			fields:  fields{"qcow2", []byte{'Q', 'F', 'I', 0xfb}, 0, 24, 8},
-			args:    args{qcowbyte},
-			want:    3544391413610329398,
-			wantErr: false,
-		},
-		{
-			name:    "does not implement size",
-			fields:  fields{"gz", []byte{0x1F, 0x8B}, 0, 0, 0},
-			args:    args{[]byte{0x1F, 0x8B}},
-			want:    0,
-			wantErr: false,
-		},
-	}
-	for _, tt := range tests {
-		t.Run(tt.name, func(t *testing.T) {
-			h := Header{
-				Format:      tt.fields.Format,
-				magicNumber: tt.fields.magicNumber,
-				mgOffset:    tt.fields.mgOffset,
-				SizeOff:     tt.fields.SizeOff,
-				SizeLen:     tt.fields.SizeLen,
-			}
-			got, err := h.Size(tt.args.b)
-			if (err != nil) != tt.wantErr {
-				t.Errorf("Header.Size() error = %v, wantErr %v", err, tt.wantErr)
-				return
-			}
-			if got != tt.want {
-				t.Errorf("Header.Size() = %v, want %v", got, tt.want)
-			}
-		})
-	}
-}
+	table.DescribeTable("Header size", func(fields fields, args args, want int64, wantErr bool) {
+		h := Header{
+			Format:      fields.Format,
+			magicNumber: fields.magicNumber,
+			mgOffset:    fields.mgOffset,
+			SizeOff:     fields.SizeOff,
+			SizeLen:     fields.SizeLen,
+		}
+		got, err := h.Size(args.b)
+		if wantErr {
+			Expect(err).To(HaveOccurred())
+		} else {
+			Expect(err).ToNot(HaveOccurred())
+		}
+		Expect(got).To(Equal(want))
+	},
+		table.Entry("get size of qcow2",
+			fields{"qcow2", []byte{'Q', 'F', 'I', 0xfb}, 0, 24, 8},
+			args{qcowbyte},
```

Comment:
same, seems we should just pass the byte array as parameter, instead of an args struct.